Control/Monad/Writer/Class.hs

```diff
@@ -204,3 +208,18 @@ instance MonadWriter w m => MonadWriter w (Strict.StateT s m) where
     tell   = lift . tell
     listen = Strict.liftListen listen
     pass   = Strict.liftPass pass
+
+#if MIN_VERSION_transformers(0,5,3)
+-- | @since 2.3
+instance
+  ( Monoid w
+  , MonadWriter w m
+#if !MIN_VERSION_base(4,8,0)
+  , Functor m
+#endif
+  ) => MonadWriter w (AccumT w m) where
```

Comment:
Per the discussion in #44, there is a design choice to be made about how the `MonadWriter` instance for `AccumT` works. It could either:

1. Lift the operations to the inner monad, or
2. Handle the operations itself.

This PR chooses approach (1), which I think is fine. But it a choice nonetheless, and I think it should be documented to avoid any confusion on the topic.

Reply:
Pushed some docs for this, let me know what you think about the wording/if anything is missing